### PR TITLE
Vendor gardener v1.83.2

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/platform azure

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other dependency
Vendor gardener/gardener =>  `v1.83.2`
```

### CODE REVIEW

File: cmd/gardener-extension-provider-azure/app/app.go [hunk 1]
```diff
@@ -202,8 +203,17 @@ func NewControllerManagerCommand(ctx context.Context) *cobra.Command {
 				return fmt.Errorf("could not update manager scheme: %w", err)
 			}
 
-			// add common meta types to schema for controller-runtime to use v1.ListOptions
-			metav1.AddToGroupVersion(scheme, machinev1alpha1.SchemeGroupVersion)
```
Comment (shafeeqes) [inline cmd/gardener-extension-provider-azure/app/app.go:206]: Don't we still need this?

File: pkg/controller/healthcheck/add.go [hunk 2]
```diff
@@ -113,14 +111,6 @@ func RegisterHealthChecks(ctx context.Context, mgr manager.Manager, opts healthc
 		workerConditionTypesToRemove = sets.New(gardencorev1beta1.ShootControlPlaneHealthy)
```
Comment (shafeeqes) [inline pkg/controller/healthcheck/add.go:111]: These variables can be inlined now.

File: pkg/controller/infrastructure/infraflow/shared/basic_context.go [hunk 3]
```diff
@@ -136,19 +136,13 @@ func (c *BasicFlowContext) AddTask(g *flow.Graph, name string, fn flow.TaskFn, o
...
 	if allOptions.Timeout > 0 {
 		tunedFn = tunedFn.Timeout(allOptions.Timeout)
 	}
 	task := flow.Task{
-		Name: name,
-		Fn:   c.wrapTaskFn(g.Name(), name, tunedFn),
-	}
+		Name:   name,
+		Fn:     c.wrapTaskFn(g.Name(), name, tunedFn),
+		SkipIf: allOptions.DoIf != nil && !*allOptions.DoIf}
```
Comment (shafeeqes) [inline pkg/controller/infrastructure/infraflow/shared/basic_context.go:145]: Nit
```suggestion
		SkipIf: allOptions.DoIf != nil && !*allOptions.DoIf
    }
```

File: pkg/controller/worker/machines_test.go [hunk 4]
```diff
@@ -66,7 +65,7 @@ var _ = Describe("Machines", func() {
 		chartApplier = mockkubernetes.NewMockChartApplier(ctrl)
 		statusWriter = mockclient.NewMockStatusWriter(ctrl)
 
-		// Let the client always the mocked status writer when Status() is called.
+		// Let the seedClient always the mocked status writer when Status() is called.
```
Comment (shafeeqes) [inline pkg/controller/worker/machines_test.go:68]:
```suggestion
		// Let the seed client always return the mocked status writer when Status() is called.
```
Similarly at other places.

File: cmd/gardener-extension-provider-azure/app/app.go (hunk 1 above)
Comment (kon-angelo) [reply cmd/gardener-extension-provider-azure/app/app.go:206]: I am actually a little bit confused about it but doesn't https://github.com/gardener/gardener-extension-provider-azure/blob/de1f4065099e7556d383cfe4787d162837ba2e12/cmd/gardener-extension-provider-azure/app/app.go#L198 already achieve the same ?

Comment (shafeeqes) [reply cmd/gardener-extension-provider-azure/app/app.go:206]: That's true. Now, I'm not sure either. Maybe you can test it if it works this way, all the other extensions also can drop this lines then.

**Comment by kon-angelo** [comment]
@shafeeqes PTAL :)

File: cmd/gardener-extension-provider-azure/app/app.go (hunk 1 above)
Comment (kon-angelo) [reply cmd/gardener-extension-provider-azure/app/app.go:206]: I didn't see any issue so far in my tests but i will keep an eye

File: pkg/controller/worker/machines_test.go [hunk 5]
```diff
@@ -65,7 +65,7 @@ var _ = Describe("Machines", func() {
 		chartApplier = mockkubernetes.NewMockChartApplier(ctrl)
 		statusWriter = mockclient.NewMockStatusWriter(ctrl)
 
-		// Let the seedClient always the mocked status writer when Status() is called.
+		// Let the seed client always the mocked status writer when Status() is called.
```
Comment (shafeeqes) [inline pkg/controller/worker/machines_test.go:68]: Nit
```
		// Let the seed client always return the mocked status writer when Status() is called.
```

**Review by shafeeqes — COMMENTED** [review]
Thanks, just a small nit left.

File: pkg/controller/worker/machines_test.go (hunk 5 above)
Comment (kon-angelo) [reply pkg/controller/worker/machines_test.go:68]: Missed that one 👍

**Review by MartinWeindel — APPROVED** [review]
/lgtm

**Pull request merged**